OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs

```diff
@@ -25,6 +25,25 @@ protected Actor FindClosestEnemy(Squad owner)
 		{
 			return owner.SquadManager.FindClosestEnemy(owner.Units.First().CenterPosition);
 		}
+
+		protected Actor GetRandomValuableTarget(Squad owner)
+		{
+			var manager = owner.SquadManager;
+			var mustDestroyedEnemy = manager.World.ActorsHavingTrait<MustBeDestroyed>(t => t.Info.RequiredForShortGame)
+					.Where(a => manager.IsPreferredEnemyUnit(a) && manager.IsNotHiddenUnit(a)).ToArray();
+
+			if (!mustDestroyedEnemy.Any())
+				return FindClosestEnemy(owner);
+
+			return mustDestroyedEnemy.Random(owner.World.LocalRandom);
```

Comment:
Is there a reason you take a random valuable target and not the closest valuable target? In my tests I've seen squads go to one side of the map to target a specific enemy, and when that building gets destroyed half-way (not uncommon in an FFA) go to the other side of the map to attack a completely different enemy.

Reply:
Because this is just decide the main direction of force processing,  choose a random actor won't hurt,  and **performance friendly** since I decided to cut down performance cost.

Reply:
It is suggest to make `attackScanRadius` larger by solving this issue.

Reply:
The problem is that you would have to make `attackScanRadius` so large that it defeats the whole purpose of the attackmovestate. I've seen this happen on big maps when the squad is nowhere near the target yet.

What I would suggest instead is to only pick a random target on the first tick after the squad is created. If the target becomes invalidated later on, choose the target that is closest to the previous target. This way the squad doesn't look so indecisive.

Reply:
Good idea on closet to dead target, but I will still insist on using a random target,  which will make AI attack unpredictable, like the same squad can first assault on playerA sub base, and then they are heading for playerB's main base. 

guess we can do an option for user in the future.

Reply:
And first target for Groundstate is granted by `SquadManager`, for both assault squad and rush squad. The difference is assault squad always choose the closet target to base, while rush squad will target enemy conyard.